graspy/inference/latent_distribution_test.py

```diff
@@ -281,23 +299,60 @@ def _sample_modified_ase(self, X, Y, pooled=False):
         # return the embeddings in the appropriate order
         return (Y, X_sampled) if reverse_order else (X_sampled, Y)
 
-    def fit(self, A1, A2):
+    def fit(self, A1, A2, pass_graph=True):
         """
         Fits the test to the two input graphs
 
         Parameters
         ----------
-        A1, A2 : nx.Graph, nx.DiGraph, nx.MultiDiGraph, nx.MultiGraph, np.ndarray
-            The two graphs to run a hypothesis test on.
+        A1, A2 : variable (see description)
+            The two graphs, or their embeddings to run a hypothesis test on.
+            Expected variable type and shape depends on input_graph attribute:
+
+            - input_graph=True
+                expects two unembedded graphs either as networkX objects, or as
+                two np.ndarrays, representing the adjacency matrices. In this
+                case will be embedded using adjacency spectral embedding.
+            - input_graph-False
+                expects two already embedded graphs. In this case they must be
+                arrays of shape (n, d) and (m, d), where d, the number of
+                components, must be shared.
+
+            Note that regardless of how the graphs are passed, they need not to
+            have the same number of vertices.
 
         Returns
         -------
         self
         """
-        A1 = import_graph(A1)
-        A2 = import_graph(A2)
+        if self.input_graph:
+            A1 = import_graph(A1)
+            A2 = import_graph(A2)
 
-        X1_hat, X2_hat = self._embed(A1, A2)
+            X1_hat, X2_hat = self._embed(A1, A2)
+        else:
+            X1_hat, X2_hat = np.array(A1), np.array(A2)
```

Comment:
probs needs to be a type check in here somewhere. consider using sklearn check_array https://scikit-learn.org/stable/modules/generated/sklearn.utils.check_array.html

Reply:
this tries casting it as well, and even kind of succeeds.

```
In [6]: G = nx.from_numpy_matrix(np.random.binomial(1, 0.5, (50, 50)))                                                

In [7]: np.array(G)                                                                                                   
Out[7]: 
array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33,
       34, 35, 36, 37, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49])

In [8]: from sklearn.utils import check_array                                                                         

In [9]: check_array(G)                                                                                                
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-9-7c9c5851fad0> in <module>
----> 1 check_array(G)

~/.virtualenvs/graphs/lib/python3.7/site-packages/sklearn/utils/validation.py in check_array(array, accept_sparse, accept_large_sparse, dtype, order, copy, force_all_finite, ensure_2d, allow_nd, ensure_min_samples, ensure_min_features, warn_on_dtype, estimator)
    554                     "Reshape your data either using array.reshape(-1, 1) if "
    555                     "your data has a single feature or array.reshape(1, -1) "
--> 556                     "if it contains a single sample.".format(array))
    557 
    558         # in the future np.flexible dtypes will be handled like object dtypes

ValueError: Expected 2D array, got 1D array instead:
array=[ 0  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23
 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47
 48 49].
Reshape your data either using array.reshape(-1, 1) if your data has a single feature or array.reshape(1, -1) if it contains a single sample.
```

Reply:
hwat do u think if i first check for isinstance of nx.Graph, raise an error saying input_graph should be false, if that succeeds, and then do the sklearn check array?

Reply:
*input graph should be true

Reply:
Sure. Seems good to check for that one explicitly cause easy mistake to make. Sklearn check array does expect an array, but does some useful checks on arrays. So do still need to do a type check in general.

Reply:
is there a safe way of checking for array-ness, then? the reason why i casted instead of checking isinstance(n.ndarray) is because i wanted to allow lists and sparse array and anything of similar type

Reply:
personally i prefer to be throwing a custom ndim exception, rather than sklearn's value error above, so i think i prefer to be checking ndim myself. in that case, it seems that all that check_array is doing is ensuring that there aren't any inifinite values, so i don't see much reason to use it, personally.

Reply:
check_array allows lists + sparse as inputs and casts to an nd.array if it is not an array.

Reply:
also you can enforce 2d

Reply:
> also you can enforce 2d

my point was precisely that i want to be doing that myself and throw a custom error, rather than generic one of sklearn

Reply:
anywhere we can get rid of input checking code that just looks like 
`if bad: 
     raise BadError()`
I am for it, because we just end up with half the library being input checking. and it is easy for us to forget stuff like checking for nan values, for instance

Reply:
> anywhere we can get rid of input checking code that just looks like
> `if bad: raise BadError()`
> I am for it, because we just end up with half the library being input checking. and it is easy for us to forget stuff like checking for nan values, for instance

i.e. you prefer me to change
`X1_hat = check_array(A1, ensure_2d=False, allow_nd=True)`
to 
`X1_hat = check_array(A1, ensure_2d=True, allow_nd=False)`
and remove explicit checks below?

Reply:
@bdpedigo just a reminder to see ^